api/trace/api.go

```diff
@@ -142,16 +142,36 @@ type Link struct {
 
 // SpanKind represents the role of a Span inside a Trace. Often, this defines how a Span
 // will be processed and visualized by various backends.
-type SpanKind string
+type SpanKind int
 
 const (
-	SpanKindInternal SpanKind = "internal"
-	SpanKindServer   SpanKind = "server"
-	SpanKindClient   SpanKind = "client"
-	SpanKindProducer SpanKind = "producer"
-	SpanKindConsumer SpanKind = "consumer"
+	// These match the proto definition, see opentelemetry/proto/trace/v1/trace.proto
```

Comment:
Doews matching this with the proto specs helps us in any way? 

Asking this because the otel specs doesn't include the `unspecified` and in the end we convert any `unspecified` to `internal` anyway.

Reply:
I wanted to avoid the default 0 value being a valid span kind. I've added a comment to indicate that the unspecified value is invalid, and the helper you suggested. It's just a convenience that these match the proto, I'll add.